src/graphql/schema.js

```diff
@@ -272,8 +307,26 @@ export function nullable(elementType: Typename): PrimitiveTypeAnnotation {
   return {nonNull: false, elementType};
 }
 
-export function nested(eggs: {
-  +[Fieldname]: PrimitiveFieldType | NodeFieldType,
-}): NestedFieldType {
-  return {type: "NESTED", eggs: {...eggs}};
+export function nested(
+  eggs: {
+    +[Fieldname]: PrimitiveFieldType | NodeFieldType,
+  },
+  fidelity?: Fidelity = faithful()
+): NestedFieldType {
+  return {type: "NESTED", eggs: {...eggs}, fidelity};
```

Comment:
Can you elaborate on this a little bit? Do you mean to add a test: `if (a.fidelity !== "FAITHFUL") throw new Error()` to `is deep-equal to but deep-distinct from its input`?

Reply:
When the user passes a value into the “constructor-like” functions in
this module, we always shallow-copy that value. This is why we create a
new object in `nested` rather than using the original `eggs`, and why in
`schema` we create new objects for each of the field types.

Just as `nested` shallow-copies its eggs, so too should it shallow-copy
its fidelity. Same for `node`. You probably want to define a helper
function `_cloneFidelity` to handle the `switch`-`case`ing. You need not
test this helper function, though it should be tested indirectly.

Does that make sense?